### PR TITLE
Issue #SH-3 feat: Separation of create tab into two sub-tabs and taking TPD as default framework for all type of creation in training sub-tab

### PR DESCRIPTION
**Pull request by sudip-mukherjee**

### CODE REVIEW

File: src/app/resourcebundles/data/creation/en.properties [hunk 1]
```diff
@@ -173,6 +173,8 @@ frmelmnts.snav.shareViaLink=Shared via link
 frmelmnts.snav.allmycontents = All My Content
 frmelmnts.snav.flagReviewer = Flag Review
 frmelmnts.snav.Collaboratingon = Collaborations
+frmelmnts.snav.textbook = textbook
```
Comment (harishgilimi) [inline src/app/resourcebundles/data/creation/en.properties:176]: Keep it consistent Make textbook to Textbook

File: src/app/client/src/app/modules/workspace/components/create-training/create-training.component.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
+<div class="ui grid mt-5">
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/workspace/components/create-training/create-training.component.html:1]: changes required for telemetry interact events pageId

File: src/app/client/src/app/modules/workspace/components/create-training/create-training.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateTrainingComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+    it('should set the role for course creation and associated resource creation', () => {
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/workspace/components/create-training/create-training.component.spec.ts:47]: Test case is missing for impression event which is set in the component

File: src/app/client/src/app/modules/workspace/workspace-routing.module.ts [hunk 4]
```diff
@@ -46,40 +46,84 @@ const routes: Routes = [
...
+              { label: 'Profile', url: '/profile' }, { label: 'My Workspace', url: '' }]
+            }
+          }
+        ]
+      },
+      {
+        path: 'create/training', component: CreateTrainingComponent, canActivate: [AuthGuard],
+        data: {
+          telemetry: {
+            env: telemetryEnv, pageid: 'workspace-content-create', uri: '/workspace/content/create/training',
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/workspace/workspace-routing.module.ts:84]: pageid should be workspace-training-content-create and same for other routes as well

File: src/app/client/src/app/modules/workspace/components/data-driven/data-driven.component.ts [hunk 5]
```diff
@@ -316,12 +317,13 @@ export class DataDrivenComponent extends WorkSpace implements OnInit, OnDestroy,
...
     * Issue #SB-1448,  If previous url is not from create page, redirect current page to 'workspace/content/create'
   */
-  checkForPreviousRouteForRedirect() {
-    const previousUrlObj = this.navigationHelperService.getPreviousUrl();
-    if (previousUrlObj && previousUrlObj.url && (previousUrlObj.url !== '/workspace/content/create')) {
-      this.redirect();
-    }
+ checkForPreviousRouteForRedirect() {
+  const previousUrlObj = this.navigationHelperService.getPreviousUrl();
+  if (previousUrlObj && previousUrlObj.url &&
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/workspace/components/data-driven/data-driven.component.ts:322]: change this condition to minimize the code and maximize the readability
```
let url = _.get(previousUrlObj, 'url')
let  routes =[ '/workspace/content/create', '/workspace/content/create/training']
if(url && _.indexOf(routes, url) === -1){
     this.redirect();
}
```

File: src/app/resourcebundles/data/creation/en.properties (hunk 1 above)
Comment (sudip-mukherjee) [reply src/app/resourcebundles/data/creation/en.properties]: done

File: src/app/client/src/app/modules/workspace/components/create-training/create-training.component.html (hunk 2 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/workspace/components/create-training/create-training.component.html:1]: done

File: src/app/client/src/app/modules/workspace/components/create-training/create-training.component.spec.ts (hunk 3 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/workspace/components/create-training/create-training.component.spec.ts:48]: added test cases

File: src/app/client/src/app/modules/workspace/workspace-routing.module.ts (hunk 4 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/workspace/workspace-routing.module.ts]: done

File: src/app/client/src/app/modules/workspace/components/data-driven/data-driven.component.ts (hunk 5 above)
Comment (sudip-mukherjee) [reply src/app/client/src/app/modules/workspace/components/data-driven/data-driven.component.ts]: done

File: src/app/client/src/app/modules/workspace/components/create-training/create-training.component.html (hunk 2 above)
Comment (harishgilimi) [reply src/app/client/src/app/modules/workspace/components/create-training/create-training.component.html:1]: This is not done  still page id doesn't have training prefix

File: src/app/client/src/app/modules/workspace/components/create-training/create-training.component.spec.ts (hunk 3 above)
Comment (harishgilimi) [reply src/app/client/src/app/modules/workspace/components/create-training/create-training.component.spec.ts:48]: There is not telemetry test case

**Pull request merged**